test/integration/associations/belongs-to-many.test.js

```diff
@@ -470,14 +470,7 @@ describe(Support.getTestDialectTeaser('BelongsToMany'), function() {
         }
       });
 
-      var self = this;
-      return this.UserTask.sync({ force: true })
-      .then(function() {
-        return self.user.setActiveTasks(self.tasks);
-      })
-      .then(function() {
-        return expect(self.user.countActiveTasks({})).to.eventually.equal(1);
-      });
+      return expect(this.user.countActiveTasks({})).to.eventually.equal(1);
```

Comment:
@sushantdhiman Can you explain what this change did? Was it a fix of the test, unrelated to the `foreignKey`/`as` change?